docker/Dockerfile

```diff
@@ -28,6 +33,7 @@ MAINTAINER FIWARE IoTAgent Team. Telefónica I+D
 # replace the whole RUN statement by the following COPY statement in your local source using :
 #
 # COPY . /opt/iotajson/
+# COPY ./docker/entrypoint.sh/
```

Comment:
Missing space and extraneous slash - shouldn't it be:

```
# COPY . /docker/entrypoint.sh
```

Reply:
I do not understand Run mv, should ENTRYPOINT be  ["/docker/entrypoint.sh"] ??

Reply:
Your `ENTRYPOINT` expects a file to be found in root `./entrypoint.sh`

The commented out copy statement copies the file to root (for a local build)

```Dockerfile
# COPY . /docker/entrypoint.sh
````

The line https://github.com/telefonicaid/iotagent-json/blob/0fbd002/docker/Dockerfile#L56

```bash
mv "${GITHUB_REPOSITORY}-${RELEASE}" /opt/iotajson && \
```

Moves all of the contents of the cloned repo into  `/opt/iotajson` - you therefore need to move `/opt/iotajson/docker/entrypoint.sh` to `/entrypoint.sh` otherwise the `Dockerfile` won't start.

The basic question is what currently happens when you build and run the `Dockerfile`?  How are you testing this change? How are you verifying that it works correctly?


For more background on the Dockerfile and how it **currently works** see:

* https://github.com/telefonicaid/iotagent-ul/pull/310
* https://github.com/telefonicaid/iotagent-json/issues/342

Reply:
When I am testing on my local the Dockerfile gives a following error-
![2019-02-08_115948](https://user-images.githubusercontent.com/42174460/52462750-67ec2d00-2b9a-11e9-8f36-1145e3eeda76.png)

Reply:
In the current Dockerfile, the ENTRYPOINT is ["bin/iotagent-json"] as all the contents of cloned repo are into /opt/iotajson therefore /bin/iotagent-json are also cloned into dis repo. Similarly when /docker/entrypoint.sh will be merged in master, it will also be part of cloned repo inside /opt/iotajson and the ENTRYPOINT ["/docker/entrypoint.sh"] will start. What do you think @jason-fox  ?

Reply:
Whilst testing use build args - from the docs:

>* To download code from your own fork of the GitHub repository add the `GITHUB_ACCOUNT` and `GITHUB_REPOSITORY` arguments to the docker build command.
>
>```console
>docker build -t iot-agent . --build-arg GITHUB_ACCOUNT=<your account> --build-arg >GITHUB_REPOSITORY=<your repo>
>```

When deciding where the `entrypoint.sh` file needs to be follow common best practice.

Looking at 3 common databases (Mongo-DB, Postgres and MySQL) they all place the 
`docker-entrypoint.sh` relatively.

```Docker
ENTRYPOINT ["/docker/entrypoint.sh"]
```
is **always** looking for root

```Docker
ENTRYPOINT ["docker/entrypoint.sh"]
```

would be relative.

Reply:
Fixes https://github.com/telefonicaid/iotagent-json/pull/380/commits/9bc569e10bd0d96ad23c10d3b6bff090cb71f2df

Reply:
I assume the comment in line 36 is no longer necessary and can be removed.

Reply:
Fixed by https://github.com/telefonicaid/iotagent-json/pull/380/commits/a320676d7372bc4db45c6baba501e010b1ccc6c7

Reply:
:+1: LGTM